types/adm-zip/index.d.ts

```diff
@@ -282,6 +363,160 @@ declare namespace AdmZip {
+        ZIP64EXTRA: 56; // extensible data sector
+
+        /* Compression methods */
+        STORED: 0; // no compression
+        SHRUNK: 1; // shrunk
+        REDUCED1: 2; // reduced with compression factor 1
+        REDUCED2: 3; // reduced with compression factor 2
+        REDUCED3: 4; // reduced with compression factor 3
+        REDUCED4: 5; // reduced with compression factor 4
+        IMPLODED: 6; // imploded
+        // 7 reserved for Tokenizing compression algorithm
+        DEFLATED: 8; // deflated
+        ENHANCED_DEFLATED: 9; // enhanced deflated
+        PKWARE: 10; // PKWare DCL imploded
+        // 11 reserved by PKWARE
+        BZIP2: 12; //  compressed using BZIP2
+        // 13 reserved by PKWARE
+        LZMA: 14; // LZMA
+        // 15-17 reserved by PKWARE
+        IBM_TERSE: 18; // compressed using IBM TERSE
+        IBM_LZ77: 19; // IBM LZ77 z
+        AES_ENCRYPT: 99; // WinZIP AES encryption method
+
+        /* General purpose bit flag */
+        // values can obtained with expression 2**bitnr
+        FLG_ENC: 1; // Bit 0: encrypted file
+        FLG_COMP1: 2; // Bit 1, compression option
+        FLG_COMP2: 4; // Bit 2, compression option
+        FLG_DESC: 8; // Bit 3, data descriptor
+        FLG_ENH: 16; // Bit 4, enhanced deflating
+        FLG_PATCH: 32; // Bit 5, indicates that the file is compressed patched data.
+        FLG_STR: 64; // Bit 6, strong encryption (patented)
+        // Bits 7-10: Currently unused.
+        FLG_EFS: 2048; // Bit 11: Language encoding flag (EFS)
+        // Bit 12: Reserved by PKWARE for enhanced compression.
+        // Bit 13: encrypted the Central Directory (patented).
+        // Bits 14-15: Reserved by PKWARE.
+        FLG_MSK: 4096; // mask header values
+
+        /* Load type */
+        FILE: 2;
+        BUFFER: 1;
+        NONE: 0;
+
+        /* 4.5 Extensible data fields */
+        EF_ID: 0;
+        EF_SIZE: 2;
+
+        /* Header IDs */
+        ID_ZIP64: 0x0001;
+        ID_AVINFO: 0x0007;
+        ID_PFS: 0x0008;
+        ID_OS2: 0x0009;
+        ID_NTFS: 0x000a;
+        ID_OPENVMS: 0x000c;
+        ID_UNIX: 0x000d;
+        ID_FORK: 0x000e;
+        ID_PATCH: 0x000f;
+        ID_X509_PKCS7: 0x0014;
+        ID_X509_CERTID_F: 0x0015;
+        ID_X509_CERTID_C: 0x0016;
+        ID_STRONGENC: 0x0017;
+        ID_RECORD_MGT: 0x0018;
+        ID_X509_PKCS7_RL: 0x0019;
+        ID_IBM1: 0x0065;
+        ID_IBM2: 0x0066;
+        ID_POSZIP: 0x4690;
+
+        EF_ZIP64_OR_32: 0xffffffff;
+        EF_ZIP64_OR_16: 0xffff;
+        EF_ZIP64_SUNCOMP: 0;
+        EF_ZIP64_SCOMP: 8;
+        EF_ZIP64_RHO: 16;
+        EF_ZIP64_DSN: 24;
+    }
+
+    interface InitOptions {
+        /* If true it disables files sorting */
+        noSort: boolean;
+        /* Read entries during load (initial loading may be slower) */
+        readEntries: boolean;
+        /* Read method */
+        method: keyof Constants;
```

Comment:
Sorry, I think I'm missing something - so the `method` could be `LOCLEN` of `EF_SIZE`, for example?

Reply:
I wanted to type the input for `method`. I gathered from [this line](https://github.com/cthackers/adm-zip/blob/bebbabf600a6ae5d64ee31cbc89a1274bd1d8c1f/adm-zip.js#L15) that the input for methods is a bunch of integers (it defaults to `0` for instance). 

I realise now that I should have made `Constants` into an Enum and used that for typing `method`. I'll make that change if it makes sense for you.

Reply:
All the values encoded in the `Constants` come from [here](https://github.com/cthackers/adm-zip/blob/master/util/constants.js) by the way.

Reply:
Weird, but it matches the underlying library.  I think the interface (rather than the enum) was the way to go and the parameter type should be either `number` or a union `Constants.x | Constants.y | ...`.

Reply:
I've reverted it to an interface, with the method type becoming `number | Constants[keyof Constants]`. I've attached the interface to an export from `adm-zip/util`, so that should work out with how the library is setup.